instagram/src/App.js

```diff
@@ -57,17 +56,24 @@ class App extends Component {
     }
   }
 
+  handleLogout = () => {
+    localStorage.clear();
+    alert('Logged Out!');
+    window.location.reload();
+  }
+
   render() {
     return (
       <div className="App">
-        <SearchBar 
+        <PostPage 
           value={this.state.searchText}
           handleChanges={this.handleChanges}
+          posts={this.state.posts}
+          handleLogout={this.handleLogout}
         />
-        <PostContainer posts={this.state.posts}/>
       </div>
     );
   }
 }
 
-export default App;
\ No newline at end of file
+export default authenticate(App)(Login);
```

Comment:
this it another way of accomplishing the HOC authentication that actually looks more similar to how I was taught as a student with web14, so I can assume that Dustin keeps refining his method for teaching this for future cohorts. well done